### DISCUSSION TITLE
Draw handles for custom script shape2d member in editor scene view

### DISCUSSION THREAD

**Discussion by hsandt** [godotengine/godot-proposals#6568] [Editor]
I added my own shape2d member to a custom script as exported var, so I can edit it numerically in the inspector. However, I cannot edit it via the usual handles in the scene, unlike native nodes having such a member.

I would need to add a CollisionShape2D to benefit from the shape editor, and reuse that shape in my code. However, I don't need the collision part so that would be a waste.

There is a similar issue in Unity where you'd often create Box and Polygon Collider components just to benefit from a shape editor, but only care about the shape itself, not the collision. However, I was also able to create more lightweight shape data members or components in this engine because I know how to draw polygon handles easily.

I'm not familiar with handle drawing in Godot. I know there is an API for this, but I'd rather avoid writing my own code for this as the native shape editor used for collisions is already pretty good, and writing a full polygon editor with filled color, etc. would be overkill for what I need.

Is there any easy way to reuse the native collision shape editor for non-collision shapes, without having to create a CollisionShape2D node?

Because if not, I'd like to submit a proposal for this. Although I don't have a clear idea of how it would look like, nor how we would code the thing (an annotation on the shape member maybe?), so I thought the discussion format may be appropriate for this.

Also applies to 3D shapes and 3D equivalent classes, although I'm not using those at the moment.

**Comment by Braveo** [comment]
Currently dealing with something similar. I've wanted to modify the NinePatchRect associated with this object since it just comes in with handles, but the issue is that having a child of a NinePatchRect is not desired for this situation so I was wondering what other solutions to this problem was other than just modifying it through code:
https://github.com/user-attachments/assets/19867cc1-81c6-447f-a5eb-8a935c680cb8

There's also another scenario where I store camera limits in a specific zone, and instead of being able to use editor handles, I modify this through a tool script with my own weird little tool:
https://github.com/user-attachments/assets/65094b34-005b-471b-b54b-59e55f9f0812

<img width="495" height="475" alt="image" src="https://github.com/user-attachments/assets/ad16ec61-2a5f-461b-ae72-e4958a496547" />
<img width="851" height="859" alt="image" src="https://github.com/user-attachments/assets/ff8d7311-0b88-4e35-ad22-f38df4da2877" />

But this tool doesn't really hold well to extending it to anything else or being decoupled from the script, so I'm just kinda stuck with modifying numbers through the inspector for a lot of other scripts. Maybe an editor plugin would work but it'd definitely be nice to see more nodes with this level of customization, such as exposing some children to be editable by default for instanced scenes (so I can modify collision shape / NinePatchRect individually per object). 

I understand why it hasn't been completely discussed since there are a lot of ways around this but it'd definitely be nice to have! Polygon drawing and other types of tools to implement custom behavior with tool scripts for. Maybe like, a way to annotate such tool script with a preset behavior like rect or point handles. Maybe the way Blender does it with geometry nodes. Or the way Unity does it (I actually haven't used handles in unity so I'm not sure). Maybe a way to get each tool to just implement a handle and get the handle's data depending on the handle type.

P.S. a NinePatchSprite would be really nice too. (I know there's a PR out of this and it's kinda off topic so I'll leave it as that)
